cypress/integration/console/auth-token-handling.spec.js

```diff
@@ -0,0 +1,112 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import stringToHash from '../../../pkg/webui/lib/string-to-hash'
+
+describe('Auth token handling', () => {
```

Comment:
1.  Yeah, these tests do not cover the particular failure scenario fixed here, which is somewhat hard to recreate (and to detect as well), should have made that clearer.
2.  There are to some extend UI tests, since they test whether the UI behaves correctly in these scenarios (no wrong redirects, no error messages). These tests are not connected to a concrete user flow, since they're basically over-arching functionality of the frontend. It only tests the behavior in some rare but possible states that the application can get in.

Reply:
If you prefer, I can remove the tests and we can discuss this in a separate PR.

Reply:
Im fine merging the fix as it is right now. But for the tests I would really prefer updating our testing guidelines to account for such scenarios.